### DISCUSSION TITLE
What happens to JobRunr (MongoDB) during e.g. leader election or temporary connection glitches?

### DISCUSSION THREAD

**Discussion by johanhaleby** [jobrunr/jobrunr#626] [Q&A]
I'm trying to understand what's happening with JobRunr (non pro version) if the connection goes down temporarily to MongoDB. I'm reading at https://www.jobrunr.io/en/documentation/pro/database-fault-tolerance/ but I'm not really sure I understand things correctly.  Let's say that MongoDB goes into leader election mode and elects a new leader, will JobRunr not continue executing if this is happening? 

The docs also says that: 
> This can off-course be monitored via the dashboard and health endpoints.

Where can I find these "health enpoints"? I've tried search the docs for them, but I cannot seem to find it.

**Comment by rdehuyss** [comment]
Hi @johanhaleby ,

If the DB is not reachable (e.g. exceptions happen when reading from / writing to the DB), JobRunr stops processing as stated in the page you mention.

JobRunr by default participates in Spring Boot / Micronaut / Quarkus health. You can also enable MicroMeter integration manually if you want. See the [JobRunrConfiguration](https://github.com/jobrunr/jobrunr/blob/master/core/src/main/java/org/jobrunr/configuration/JobRunrConfiguration.java#L293-L302).

**Reply by johanhaleby** [reply]
@rdehuyss Thanks for your reply. 

Does "stop processing" mean the equivalent of calling `JobRunr.destroy()`? Or more specifically, how would one, using the API,  "resume processing" if this happens? What micrometer "events" can one check to see if jobrunr has "stopped processing"?
